go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -402,6 +445,32 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 			continue
 		}
 
+		network := networkMap[nadName]
+		hasIPAMClaims := network.IPAMClaimReference != ""
+		if hasIPAMClaims {
+			ipamClaim, err := bsnc.ipamClaimsReconciler.FindIPAMClaim(network.IPAMClaimReference, network.Namespace)
+			if err == nil {
+				hasIPAMClaims = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
+			} else if apierrors.IsNotFound(err) {
+				klog.Errorf("Failed to retrieve IPAMClaim %q but will release IPs: %v", network.IPAMClaimReference, err)
+				hasIPAMClaims = false
+			} else {
+				return fmt.Errorf(
+					"failed to find IPAMClaim %q",
+					fmt.Sprintf("%s/%s", network.Namespace, network.IPAMClaimReference),
+				)
+			}
+		}
+		if hasIPAMClaims {
+			klog.Infof(
+				"will not release IPs for pod %s/%s, ips: %s network %s because it has IPAM claims",
+				pod.Namespace,
+				pod.Name,
+				util.JoinIPNetIPs(pInfo.ips, " "),
+				bsnc.GetNetworkName(),
+			)
+			continue
+		}
```

Comment:
Any reason we shouldn't be using the exact same code we have on the interconnect side:
```
	hasPersistentIPs := a.netInfo.AllowsPersistentIPs() && hasIPAM
	hasIPAMClaim := network != nil && network.IPAMClaimReference != ""
	if hasIPAMClaim && !hasPersistentIPs {
		klog.Errorf(
			"Pod %s/%s referencing an IPAMClaim on network %q which does not honor it",
			pod.GetNamespace(),
			pod.GetName(),
			a.netInfo.GetNetworkName(),
		)
		hasIPAMClaim = false
	}
	if hasIPAMClaim {
		ipamClaim, err := a.findIPAMClaim(network)
		hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
		if apierrors.IsNotFound(err) {
			klog.Errorf("Failed to retrieve IPAMClaim %q but will release IPs: %v", network.IPAMClaimReference, err)
		} else if err != nil {
			return fmt.Errorf("failed to get IPAMClaim %s/%s: %w", network.Namespace, network.IPAMClaimReference, err)
		}
	}
```

Here you log when IPs are not released, but not there. 
There you check if the network supports persitent IPs, but not here. 
Looking for consistency.

Reply:
Done - I think.